packages/api/src/controllers/task.ts

```diff
@@ -279,6 +279,34 @@ app.post(
   }
 );
 
+app.post("/:id/retry", authorizer({ anyAdmin: true }), async (req, res) => {
+  const { id } = req.params;
+  const task = await db.task.get(id, { useReplica: false });
+  if (!task) {
+    return res.status(404).json({ errors: ["task not found"] });
+  } else if (!["failed", "cancelled"].includes(task.status?.phase)) {
+    return res
+      .status(400)
+      .json({ errors: ["task is not in a retryable state"] });
+  }
+
+  const user = await db.user.get(task.userId);
+  if (!user) {
+    return res.status(500).json({ errors: ["user not found"] });
+  }
+
+  if (!req.user.admin && req.user.id !== task.userId) {
+    return res.status(403).json({
+      errors: ["users may only retry their own tasks"],
+    });
+  }
+
+  await req.taskScheduler.retryTask(task, "retry requested by user");
```

Comment:
This will not work. The task is already at its last retry and this function will also have a sleep of up to 3 minutes which is unacceptable in an API call. You need to go inside this function you are calling and make sure it does what you want.

Reply:
Also, this error message doesn't make sense The task stores the error message from the last retry that failed, while here you are replacing it with something that is not an error message. You should either keep the same message or clear it and restart the retry count to 0 (I think the ideal flow would be keeping the error message & retry count, just retrying until a higher retry count like from 3 to 6 instead of 0 to 3?)